flink/v1.15/flink/src/main/java/org/apache/iceberg/flink/sink/CommitSummary.java

```diff
@@ -0,0 +1,93 @@
+/*
+ * Licensed to the Apache Software Foundation (ASF) under one
+ * or more contributor license agreements.  See the NOTICE file
+ * distributed with this work for additional information
+ * regarding copyright ownership.  The ASF licenses this file
+ * to you under the Apache License, Version 2.0 (the
+ * "License"); you may not use this file except in compliance
+ * with the License.  You may obtain a copy of the License at
+ *
+ *   http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing,
+ * software distributed under the License is distributed on an
+ * "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY
+ * KIND, either express or implied.  See the License for the
+ * specific language governing permissions and limitations
+ * under the License.
+ */
+package org.apache.iceberg.flink.sink;
+
+import java.util.Arrays;
+import java.util.NavigableMap;
+import java.util.concurrent.atomic.AtomicLong;
+import org.apache.iceberg.io.WriteResult;
+import org.apache.iceberg.relocated.com.google.common.base.MoreObjects;
+
+class CommitSummary {
+
+  private final AtomicLong dataFilesCount = new AtomicLong();
+  private final AtomicLong dataFilesRecordCount = new AtomicLong();
+  private final AtomicLong dataFilesByteCount = new AtomicLong();
+  private final AtomicLong deleteFilesCount = new AtomicLong();
+  private final AtomicLong deleteFilesRecordCount = new AtomicLong();
+  private final AtomicLong deleteFilesByteCount = new AtomicLong();
+
+  CommitSummary(NavigableMap<Long, WriteResult> pendingResults) {
```

Comment:
Shouldn't all of these metrics come from the commit path? I think that these would be produced by core and then passed to Flink by adding a way to set the metrics context.

@nastra, what do you think?

Reply:
I think that makes a lot of sense to track those metrics in core itself and then have a particular metrics context in order to customize what type of metrics framework to use underneath.

Reply:
the problem with the `Listeners` API is that it is a global static. There is no `unregister` API. it is difficult to associate listener at table scope, as we want to have `IcebergFilesCommitter` in Flink to register listener for the table that it writes to.

It would be convenient if `PendingUpdate#commit` return a `CommitResult` (instead of `void`). What do you think of adding a new `UpdateResult PendingUpdate#apply()` method that return the action result? @rdblue @nastra

Reply:
BTW, this class also consolidate the computations of the file count in various places in the code path. FlinkSink already have the information here. Maybe we can share/reuse the `CommitSummary` class with the core module?